src/rgw/rgw_data_sync.cc

```diff
@@ -3171,26 +3176,51 @@ class InitBucketFullSyncStatusCR : public RGWCoroutine {
 
       if (info.oldest_gen == 0) {
         if (check_compat) {
-          // try to convert existing per-shard incremental status for backward compatibility
-          yield call(new CheckAllBucketShardStatusIsIncremental(sc, sync_pair, num_shards, &all_incremental));
-          if (retcode < 0) {
-            return set_cr_error(retcode);
-          }
-          if (all_incremental) {
-            // we can use existing status and resume incremental sync
-            status.state = BucketSyncState::Incremental;
-          }
-        }
+	  yield {
+	    // use shard count from our log gen=0
+	    // try to convert existing per-shard incremental status for backward compatibility
+	    const auto& log = source_info.layout.logs.front();
+	    bool no_zero = false;
+	    if (log.gen > 0) {
+	      ldpp_dout(dpp, 20)
+		<< "no generation zero when check compatibility" << dendl;
+	      no_zero = true;
+	    }
+	    if (log.layout.type != rgw::BucketLogType::InIndex) {
+	      ldpp_dout(dpp, 20) <<
+		"unrecognized log layout type when when checking compatibility " << log.layout.type << dendl;
```

Comment:
first message 'when check compatibility'
second message 'type when when checking'

nit: newline directly after `ldpp_dout(dpp, 20)` seems like a waste of vertical space